### PR TITLE
Removes some static nature of model controller

### PR DESCRIPTION
**Pull request by tlm**
### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [x] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?
 - [x] Do comments answer the question of why design decisions were made?

----

## Description of change

Removes some of the static nature out of the model agent.

## QA steps

*Please replace with how we can verify that the change works?*

cd tests && ./main.sh caasadmission

### CODE REVIEW

**Comment by tlm** [comment]
!!build!!

**Comment by tlm** [comment]
!!build!!

**Comment by tlm** [comment]
!!build!!

File: caas/kubernetes/provider/modeloperator.go [hunk 1]
```diff
@@ -151,13 +157,14 @@ func modelOperatorDeployment(
...
 	modelUUID string,
+	serviceName string,
 	volumes []core.Volume,
 	volumeMounts []core.VolumeMount,
 ) (*apps.Deployment, error) {
 
 	moLabels := modelOperatorLabels(operatorName)
 
-	jujudCmd := fmt.Sprintf("$JUJU_TOOLS_DIR/jujud model --model-uuid=%s", modelUUID)
+	jujudCmd := fmt.Sprintf("$JUJU_TOOLS_DIR/jujud model --model-uuid=%s --logging-config=\"<root>=DEBUG\"", modelUUID)
```
Comment (wallyworld) [inline caas/kubernetes/provider/modeloperator.go:167]: logging config should come from the model config setting, not hard coded
There's a logger worker which handles this. If not already part of the model operator manifold, it needs to be added

File: worker/muxhttpserver/server.go [hunk 2]
```diff
@@ -31,12 +31,22 @@ type Logger interface {
...
 type Server struct {
 	catacomb catacomb.Catacomb
+	info     ServerInfo
 	listener net.Listener
 	logger   Logger
 	Mux      *apiserverhttp.Mux
 	server   *http.Server
 }
 
+type ServerInfo interface {
```
Comment (wallyworld) [inline worker/muxhttpserver/server.go:41]: whack some comments on these

Comment (tlm) [reply worker/muxhttpserver/server.go:41]: Done & Done

File: caas/kubernetes/provider/modeloperator.go (hunk 1 above)
Comment (tlm) [reply caas/kubernetes/provider/modeloperator.go]: Done & Done

File: cmd/jujud/agent/modeloperator/manifolds.go [hunk 3]
```diff
@@ -76,22 +87,36 @@ func Manifolds(config ManifoldConfig) dependency.Manifolds {
...
 
+		// The logging config updater is a leaf worker that indirectly
+		// controls the messages sent via the log sender or rsyslog,
+		// according to changes in environment config. We should only need
+		// one of these in a consolidated agent.
+		loggingConfigUpdaterName: logger.Manifold(logger.ManifoldConfig{
+			AgentName:       agentName,
+			APICallerName:   apiCallerName,
+			LoggingContext:  loggo.DefaultContext(),
+			Logger:          loggo.GetLogger("juju.worker.logger"),
```
Comment (wallyworld) [inline cmd/jujud/agent/modeloperator/manifolds.go:98]: logger should have a different, unique name

**Comment by tlm** [comment]
$$merge$$

File: worker/muxhttpserver/server.go [hunk 4]
```diff
@@ -29,14 +29,28 @@ type Logger interface {
...
 type Server struct {
 	catacomb catacomb.Catacomb
+	info     ServerInfo
 	listener net.Listener
 	logger   Logger
 	Mux      *apiserverhttp.Mux
 	server   *http.Server
 }
 
+// ServerInfo represents a small interface for informining interested party's
```
Comment (manadart) [inline worker/muxhttpserver/server.go:43]: "_informing_ interested _parties_ through manifolds _that output_"

**Pull request merged**